internal/gatewayapi/translator.go

```diff
@@ -65,6 +66,11 @@ type Translator struct {
 	// GlobalRateLimitEnabled is true when global
 	// ratelimiting has been configured by the admin.
 	GlobalRateLimitEnabled bool
+
+	// ExtensionGroupKinds stores the group/kind for all resources
+	// introduced by an Extension so that the translator can
+	// store referenced resources in the IR for later use.
+	ExtensionGroupKinds []schema.GroupKind
```

Comment:
The `TranslationResult.Resources` only has `ExtensionRefFilters   []unstructured.Unstructured` which are just the resources from an extension that we're watching. I added this so that in the gateway API translator, we have a list of the Extension's supported groups/kinds for the call to `ValidateHTTPRouteFilter`. The `gatewayAPIReconciler` has `extGVKs   []schema.GroupVersionKind` for the same reason, but `ValidateHTTPRouteFilter` is also called from the gateway API translator which doesn't have that info currently.

Reply:
is there a need to validate twice, once in provider and another time in gateway-api translator ?